geocoding/android/src/main/java/com/baseflow/geocoding/MethodCallHandlerImpl.java

```diff
@@ -1,142 +1,173 @@
 package com.baseflow.geocoding;
 
 import android.location.Address;
+import android.os.Handler;
+import android.os.Looper;
 import android.util.Log;
+
 import androidx.annotation.Nullable;
 
 import com.baseflow.geocoding.utils.AddressMapper;
 import com.baseflow.geocoding.utils.LocaleConverter;
 
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 import io.flutter.plugin.common.BinaryMessenger;
 import io.flutter.plugin.common.MethodCall;
 import io.flutter.plugin.common.MethodChannel;
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
 import io.flutter.plugin.common.MethodChannel.Result;
 
-import java.io.IOException;
-import java.util.List;
-
 /**
  * Translates incoming Geocoding MethodCalls into well formed Java function calls for {@link
  * Geocoding}.
  */
 final class MethodCallHandlerImpl implements MethodCallHandler {
-    private static final String TAG = "MethodCallHandlerImpl";
-    private final Geocoding geocoding;
-    @Nullable private MethodChannel channel;
-
-    /** Forwards all incoming MethodChannel calls to the given {@code geocoding}. */
-    MethodCallHandlerImpl(Geocoding geocoding) {
-        this.geocoding = geocoding;
-    }
-
-    @Override
-    public void onMethodCall(MethodCall call, Result result) {
-        switch (call.method) {
-            case "locationFromAddress":
-                onLocationFromAddress(call, result);
-                break;
-            case "placemarkFromCoordinates":
-                onPlacemarkFromCoordinates(call, result);
-                break;
-            default:
-                result.notImplemented();
-                break;
+  private static final String TAG = "MethodCallHandlerImpl";
+  private final Geocoding geocoding;
+  private ExecutorService executor = Executors.newSingleThreadExecutor();
```

Comment:
I know this works but i am also wondering if we should also consider using the `TaskQueue` API. https://docs.flutter.dev/development/platform-integration/platform-channels#channels-and-platform-threading